### DISCUSSION TITLE
[API Request] Add audio metadata to song response type

### DISCUSSION THREAD

**Discussion by khers** [opensubsonic/open-subsonic-api#98] [API requests]
### Type of change

API Clarification

### Proposal description

in working on an open subsonic client for the Music Assistant(https://github.com/music-assistant/) project, I came across a user who could not stream some of their library from Navidrome (also tried Gonic with the same results) to the Music Assistant, but when the same file was offered to MA via the filesystem, everything worked fine. After some investigating, it came down to the sample rate of the audio file being 96 kHz instead of the assumed 44.1 kHz. However, aside from streaming the data and asking ffmpeg or similar  to analyze it, I don't have a way to get this information from the OS API.

I would like to see bit depth, sample rate, and the number of channels exposed along with the existing bit rate and content type for each song.

### Backward compatibility

- [X] No backward compatibility impact.

### API details

We need to complete the documentation for the song return type and as a part of that, I would like to see any of the encoding meta data that we can provide to clients as much information about the streamed data as we can.

**Comment by dweymouth** [comment]
There is a discussion open for this already, but I have to ask - in what scenario do you need to assume a sample rate? That seems unusual as most playback backends will look up the samplerate from the media file itself when playing it

**Comment by epoupon** [comment]
We already added all this in Child responses, see https://github.com/opensubsonic/open-subsonic-api/pull/83 ?

**Reply by dweymouth** [reply]
Oh yeah forgot we finalized it already - except for codec/container since that's a huge can of worms

**Reply by Tolriq** [reply]
Yes but it's needed and I'll come back to that soon :)

We need a proper transcoding API to handle devices that can't play certain things like 96/24 but only need to down to 48/24 at same quality and not low quality transcode, same for some less known codecs, ....

And so we need the codec details too.
